Libplanet.Tests/Net/Protocols/TestTransport.cs

```diff
@@ -68,47 +67,69 @@ public class TestSwarm : ISwarm
 
-        internal bool Running => !(swarmCancellationTokenSource is null);
+        internal bool Running => !(_swarmCancellationTokenSource is null);
 
         public void Dispose()
         {
         }
 
-        public async void Start()
+        public async Task StartAsync(
+            CancellationToken cancellationToken = default(CancellationToken))
         {
-            _logger.Debug("Starting swarm {Peer}.", AsPeer);
-            swarmCancellationTokenSource = new CancellationTokenSource();
-            await ProcessRuntime(swarmCancellationTokenSource.Token);
+            _logger.Debug("Starting transport of {Peer}.", AsPeer);
+            _swarmCancellationTokenSource = new CancellationTokenSource();
+            await Task.Delay(10, cancellationToken);
         }
 
-        public Task BootstrapAsync(
-            IEnumerable<BoundPeer> peers,
+        public async Task RunAsync(
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var token = cancellationToken == default(CancellationToken)
+                ? _swarmCancellationTokenSource.Token
+                : CancellationTokenSource
+                    .CreateLinkedTokenSource(
+                        _swarmCancellationTokenSource.Token, cancellationToken)
+                    .Token;
+            await ProcessRuntime(token);
+        }
+
+        public async Task StopAsync(
+            TimeSpan waitFor,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            _logger.Debug("Stopping transport of {Peer}.", AsPeer);
+            _swarmCancellationTokenSource.Cancel();
+            await Task.Delay(waitFor, cancellationToken);
+        }
+
+#pragma warning disable S4457 // Cannot split the method since method is in interface
+        public async Task BootstrapAsync(
+            IImmutableList<BoundPeer> bootstrapPeers,
             TimeSpan? pingSeedTimeout = null,
             TimeSpan? findPeerTimeout = null,
+            int depth = 3,
             CancellationToken cancellationToken = default(CancellationToken))
         {
             if (!Running)
             {
                 throw new SwarmException("Start swarm before use.");
             }
 
-            if (peers is null)
+            if (bootstrapPeers is null)
             {
-                throw new ArgumentNullException(nameof(peers));
+                throw new ArgumentNullException(nameof(bootstrapPeers));
             }
 
-            async Task DoBootstrapAsync()
-            {
-                await Protocol.BootstrapAsync(
-                    peers.ToImmutableList(),
-                    pingSeedTimeout,
-                    findPeerTimeout,
-                    Kademlia.MaxDepth,
-                    cancellationToken);
-            }
-
-            return DoBootstrapAsync();
+            await Protocol.BootstrapAsync(
+                bootstrapPeers.ToImmutableList(),
```

Comment:
Isn't this okay without `.ToImmutableList()`?

```suggestion
                bootstrapPeers,
```